eng/pipelines/templates/jobs/live.tests.yml

```diff
@@ -0,0 +1,197 @@
+      )
+
+  # Install apt dependencies (if appropriate)
+  - bash: sudo apt install -y $(AptDependencies)
+    condition: and(succeededOrFailed(), ne(variables['AptDependencies'], ''))
+    displayName: Install dependencies from apt
+
+  - template: /eng/pipelines/templates/steps/vcpkg.yml
+
+  - script: |
+      dotnet tool install -g dotnet-reportgenerator-globaltool
+      dotnet tool install dotnet-reportgenerator-globaltool --tool-path tools
+    displayName: Install coverage tools
+    # CODE_COVERAGE variable is '' (do-not-generate) in all matrix but linux-gcc
+    # It is 'enabled' by default on linux-gcc but it can be opt-out by each pipeline (disabled) 
+    condition: and(succeededOrFailed(), ne(variables['CODE_COVERAGE'], 'disabled'), ne(variables['CODE_COVERAGE'], ''))
+
+  - template: /eng/pipelines/templates/steps/cmake-build.yml
+    parameters:
+      ServiceDirectory: ${{ parameters.ServiceDirectory }}
+      GenerateArgs: $(CmakeArgs)
+      BuildArgs: "$(BuildArgs)"
+      Env: "$(CmakeEnvArg)"
+
+  - template: /eng/common/TestResources/deploy-test-resources.yml
+    parameters:
+      ServiceDirectory: ${{ parameters.ServiceDirectory }}
+      ${{ if or(parameters.Location, parameters.CloudConfig.Location) }}:
+        Location: ${{ coalesce(parameters.Location, parameters.CloudConfig.Location) }}
+      SubscriptionConfiguration: ${{ parameters.CloudConfig.SubscriptionConfiguration }}
+
+    # For non multi-config generator use the same build configuration to run tests
+    # We don't need to set it to invoke ctest
+    # Visual Studio generator used in CI is a multi-config generator.
+    # As such, it requires the configuration argument for building and invoking ctest
+  - script: ctest $(WindowsCtestConfig) -V --tests-regex "${{ parameters.CtestRegex }}" --no-compress-output -T Test
+    workingDirectory: build
+    displayName: ctest
+    # Runs only if test-resources are happly deployed.
+    # unit-tests runs for those configs where samples are not ran.
+    # This enables to run tests and samples at the same time as different matrix configuration.
+    # Then unit-tests runs, samples should not run.
+    condition: and(
+      succeeded(),
+      ne(variables['RunSamples'], '1'))
+
+  - task: PublishTestResults@2
+    inputs:
+      testResultsFormat: cTest
+      testResultsFiles: Testing/*/Test.xml
+      testRunTitle: $(Agent.JobName)
+      searchFolder: build
+      mergeTestResults: true
+      publishRunAttachments: true
+    displayName: Publish test results
+    # this step only makes sense when ctest has run
+    condition: and(succeededOrFailed(), ne(variables['RunSamples'], '1'))
+
+    # Running Samples step.
+    # Will run samples described on a file name [service]-samples.txt within the build directory.
+    # For example keyvault-samples.txt.
+    # The file is written by CMake during configuration when building samples.
+  - bash: |
```

Comment:
Similar comment here to above. If this is inline, do people have to copy this script in order to run samples tests locally? If all the logic is contained in scripts with simple calls to them in the yaml, then people can more easily test things for themselves.

Reply:
https://github.com/Azure/azure-sdk-for-cpp/issues/3565